cli/lsp/tsc.rs

```diff
@@ -2819,31 +2818,37 @@ fn op_script_version(
 /// server.
 fn js_runtime(performance: Arc<Performance>) -> JsRuntime {
   JsRuntime::new(RuntimeOptions {
-    extensions: vec![init_extension(performance)],
+    extensions: vec![deno_tsc::init_runtime(performance)],
     startup_snapshot: Some(tsc::compiler_snapshot()),
     ..Default::default()
   })
 }
 
-fn init_extension(performance: Arc<Performance>) -> Extension {
-  Extension::builder("deno_tsc")
-    .ops(vec![
-      op_is_cancelled::decl(),
-      op_is_node_file::decl(),
-      op_load::decl(),
-      op_resolve::decl(),
-      op_respond::decl(),
-      op_script_names::decl(),
-      op_script_version::decl(),
-    ])
-    .state(move |state| {
-      state.put(State::new(
-        Arc::new(StateSnapshot::default()),
-        performance.clone(),
-      ));
-    })
-    .build()
-}
+deno_core::ops!(
+  deno_ops,
+  [
+    op_is_cancelled,
+    op_is_node_file,
+    op_load,
+    op_resolve,
+    op_respond,
+    op_script_names,
+    op_script_version,
+  ]
+);
+
+deno_core::extension!(deno_tsc,
+  ops = deno_ops,
+  config = {
+    performance: Arc<Performance>
+  },
+  state = |state, performance| {
+    state.put(State::new(
+      Arc::new(StateSnapshot::default()),
+      performance.clone(),
+    ));
+  },
+);
```

Comment:
Sweet! This is exactly what I had in mind. Not sure if `config` is the name I would go with, but this is the exact idea